src/state/MediaViewModel.ts

```diff
@@ -232,18 +235,45 @@ export class LocalUserMediaViewModel extends BaseUserMediaViewModel {
  * A remote participant's user media.
  */
 export class RemoteUserMediaViewModel extends BaseUserMediaViewModel {
-  private readonly _locallyMuted = new BehaviorSubject(false);
+  private readonly locallyMutedToggle = new Subject<void>();
+  private readonly localVolumeAdjustment = new Subject<number>();
+  private readonly localVolumeCommit = new Subject<void>();
+
   /**
-   * Whether we've disabled this participant's audio.
+   * The volume to which this participant's audio is set, as a scalar
+   * multiplier.
    */
-  public readonly locallyMuted: Observable<boolean> = this._locallyMuted;
+  public readonly localVolume: Observable<number> = merge(
+    this.locallyMutedToggle.pipe(map(() => "toggle mute" as const)),
+    this.localVolumeAdjustment,
+    this.localVolumeCommit.pipe(map(() => "commit" as const)),
+  ).pipe(
+    accumulate(
+      { muted: false, volume: 1, committedVolume: 1 },
+      (state, event) =>
+        event === "toggle mute"
+          ? { ...state, muted: !state.muted }
+          : event === "commit"
+            ? { ...state, committedVolume: state.volume }
+            : // Volume adjustment
+              event === 0
+              ? // Dragging the slider to zero should have the same effect as
+                // muting: reset the volume to the committed volume, as if it were
+                // never dragged
+                { ...state, muted: true, volume: state.committedVolume }
+              : { ...state, muted: false, volume: event },
+    ),
```

Comment:
IMO using a `switch` statement reads better:
```suggestion
      (state, event) => {
        switch (event) {
        case "toggle mute":
          return { ...state, muted: !state.muted };
        case "commit":
          return { ...state, committedVolume: state.volume };
        case 0: // Volume adjustment
          // Dragging the slider to zero should have the same effect as
          // muting: reset the volume to the committed volume, as if it were
          // never dragged
          return { ...state, muted: true, volume: state.committedVolume };
        default:
          return { ...state, muted: false, volume: event };
        }
      }),
```